compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -408,11 +413,12 @@ BLangVariableReference buildStream(List<BLangNode> clauses, BType resultType, Sy
                     addStreamFunction(block, initPipeline, orderFunc);
                     break;
                 case SELECT:
-                    BLangVariableReference selectFunc = addSelectFunction(block, (BLangSelectClause) clause);
+                    BLangVariableReference selectFunc = addSelectFunction(block, (BLangSelectClause) clause,
+                            stmtsToBePropagated);
```

Comment:
Why do we have to inject namespaces explicitly into these block? How is it different from a local variable declaration declared in the same scope?

Since we have injected it only to select and do cases. If we use it in another clause, the compiler still crashes.
e.g.
```bal
public function main() {
    xmlns "http://www.so2w.org" as localNS;
    xml x = from int i in 1 ... 3 let xml y = xml `<empRecord employeeId="${i}"><localNS:id>${i}</localNS:id></empRecord>`
        select xml `<record>${y}</record>`;
}
```